src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelHelpers.cs

```diff
@@ -136,9 +136,12 @@ public static Exception ConvertHttpRequestException(HttpRequestException excepti
             var innerSocketException = exception.InnerException as SocketException;
             if (innerSocketException != null)
             {
-                var socketErrorCode = innerSocketException.SocketErrorCode;
+                SocketError socketErrorCode = innerSocketException.SocketErrorCode;
                 switch (socketErrorCode)
                 {
+                    case SocketError.TryAgain:
+                    case SocketError.NoRecovery:
+                    case SocketError.NoData:
                     case SocketError.HostNotFound:
```

Comment:
is this an exhaustive list of all socket error codes where we could have returned "endpoint not found"?

Reply:
According to the NCL team it is exhaustive at least on Windows.